pkg/query/columnquery.go

```diff
@@ -368,15 +343,31 @@ func (q *ColumnQueryAPI) Query(ctx context.Context, req *pb.QueryRequest) (*pb.Q
 	// Apply sandwich view filtering if specified
 	sandwichByFunction := req.GetSandwichByFunction()
 	if sandwichByFunction != "" {
+		// Create a stack filter for sandwich view
+		sandwichFilter := &pb.Filter{
+			Filter: &pb.Filter_StackFilter{
+				StackFilter: &pb.StackFilter{
+					Filter: &pb.StackFilter_Criteria{
+						Criteria: &pb.FilterCriteria{
+							FunctionName: &pb.StringCondition{
+								Condition: &pb.StringCondition_Contains{
+									Contains: sandwichByFunction,
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		sandwichFilters := []*pb.Filter{sandwichFilter}
```

Comment:
As a user I think I would expect sandwich filters to be the defined filters + the focused frame instead of overriding all of them.